internal/cli/clusters/upgrade.go

```diff
@@ -72,18 +102,110 @@ func (opts *UpgradeOpts) Run() error {
 	return opts.Print(r)
 }
 
+func (opts *UpgradeOpts) atlasTenantClusterUpgradeRequest20240805() (*atlasv2.AtlasTenantClusterUpgradeRequest20240805, error) {
+	var cluster *atlasv2.AtlasTenantClusterUpgradeRequest20240805
+	if opts.filename != "" {
+		err := file.Load(opts.fs, opts.filename, &cluster)
+		if err != nil {
+			return nil, err
+		}
+
+		cluster.Name = opts.name
+
+		return cluster, nil
+	}
+
+	flexClusterDescription, err := opts.store.FlexCluster(opts.ProjectID, opts.name)
+	if err != nil {
+		return nil, err
+	}
+
+	return opts.newAtlasTenantClusterUpgradeRequestFromFlexClusterDescription(flexClusterDescription), nil
+}
+
+func (opts *UpgradeOpts) newAtlasTenantClusterUpgradeRequestFromFlexClusterDescription(flexCluster *atlasv2.FlexClusterDescription20241113) *atlasv2.AtlasTenantClusterUpgradeRequest20240805 {
+	mdbVersion := flexCluster.MongoDBVersion
+	if opts.mdbVersion != "" {
+		mdbVersion = &opts.mdbVersion
+	}
+
+	terminationProtectionEnabled := flexCluster.TerminationProtectionEnabled
+	if opts.disableTerminationProtection {
+		terminationProtectionEnabled = pointer.Get(false)
+	}
+
+	if opts.enableTerminationProtection {
+		terminationProtectionEnabled = pointer.Get(true)
+	}
+
+	var tags []atlasv2.ResourceTag
+	if flexCluster.Tags != nil {
+		tags = *flexCluster.Tags
+	}
+
+	if len(opts.tag) > 0 {
+		newTags := newResourceTags(opts.tag)
+		tags = append(tags, *newTags...)
+	}
+
+	backupEnabled := false
+	if flexCluster.BackupSettings != nil {
+		backupEnabled = *flexCluster.BackupSettings.Enabled
```

Comment:
https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_5_best_practices.md#using-getters-instead-of-direct-field-access